qt/scientific_interfaces/Indirect/IndirectDataTablePresenter.cpp

```diff
@@ -0,0 +1,578 @@
+#include "IndirectDataTablePresenter.h"
+
+#include "MantidQtWidgets/Common/SignalBlocker.h"
+
+#include <boost/numeric/conversion/cast.hpp>
+
+#include <QHeaderView>
+#include <QItemDelegate>
+#include <QLineEdit>
+#include <QRegExpValidator>
+
+namespace {
+using MantidQt::CustomInterfaces::IDA::Spectra;
+using MantidQt::CustomInterfaces::IDA::DiscontinuousSpectra;
+
+namespace Regexes {
+const QString EMPTY = "^$";
+const QString SPACE = "(\\s)*";
+const QString COMMA = SPACE + "," + SPACE;
+const QString NATURAL_NUMBER = "(0|[1-9][0-9]*)";
+const QString REAL_NUMBER = "(-?" + NATURAL_NUMBER + "(\\.[0-9]*)?)";
+const QString REAL_RANGE = "(" + REAL_NUMBER + COMMA + REAL_NUMBER + ")";
+const QString MASK_LIST =
+    "(" + REAL_RANGE + "(" + COMMA + REAL_RANGE + ")*" + ")|" + EMPTY;
+} // namespace Regexes
+
+class ExcludeRegionDelegate : public QItemDelegate {
+public:
+  QWidget *createEditor(QWidget *parent, const QStyleOptionViewItem &,
+                        const QModelIndex &) const override {
+    QLineEdit *lineEdit = new QLineEdit(parent);
+    lineEdit->setValidator(
+        new QRegExpValidator(QRegExp(Regexes::MASK_LIST), parent));
```

Comment:
In this circumstance, the bare new wouldn't be different from using a unique_ptr and release -- since the owning raw-pointer would exist at the same point and in the same timeline.

Reply:
However this is an edge case, across the code base we are trying to get rid of new altogether for a few reasons detailed: https://stackoverflow.com/a/37514601

The most important ones being (in the context of the entire codebase)

> make_unique prevents the unspecified-evaluation-order leak triggered by expressions like foo(unique_ptr<X>(new X), unique_ptr<Y>(new Y)). (Following the advice "never say new" is simpler than "never say new, unless you immediately give it to a named unique_ptr".)

> make_unique is carefully implemented for exception safety and is recommended over directly calling unique_ptr constructors.

In this case it adds no advantages, however a rule of thumb with exceptions, is no longer a rule of thumb ;)